packages/compiler/index.ts

```diff
@@ -1,24 +1,29 @@
 import { createUnplugin } from 'unplugin';
-import * as babel from '@babel/core';
+import { transformAsync } from '@babel/core';
 import { plugin } from './babel';
 
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 interface UserOptions {}
 
 export const unplugin = createUnplugin((_options: UserOptions) => {
   return {
-    name: 'unplugin-million',
+    enforce: 'pre',
+    name: 'million',
     transformInclude(id: string) {
       return /\.[jt]sx$/.test(id);
     },
     async transform(code: string) {
-      const result = await babel.transformAsync(code, { plugins: [plugin] });
-      return result.code;
+      const result = await transformAsync(code, {
+        plugins: ['@babel/plugin-syntax-jsx', plugin],
```

Comment:
```suggestion
        parseOpts: {
      plugins: ['typescript', 'jsx']
    },
```

Do we only need JSX / TS parsing support?

Reply:
good idea, does babel support ts on its own? Also, when I tried adding parseOpts the types told me it wasn't a valid field